ansible/roles/fanout/tasks/fanout_sonic.yml

```diff
@@ -58,4 +58,4 @@
       include_tasks:
         sonic/fanout_sonic_202311.yml
       when: dry_run is not defined and incremental is not defined
-  when: "'2023' in fanout_sonic_version['build_version']"
+  when: "'2023' in fanout_sonic_version['build_version'] or '2024' in fanout_sonic_version['build_version']"
```

Comment:
Why does 2024 image use sonic/fanout_sonic_202311.yml?

Reply:
Currently, the sonic fanout configuration for 202311 and 202405 is the same. So use the same yaml file to handle the two branch.

Reply:
It doesn't make sense to me, if we follow this approach, we will be forced to patch `fanout_sonic_202311.yml` in the future. As time goes on, the number of patches will increase, raising the risk of importing new issues.
So I suggest to create a separate file for 202405.

Reply:
Reasonable, sure, I will have an internal discussion and update then.

Reply:
Hi @xwjiang-ms , updated the PR.